### ISSUE TITLE
TryLoadLoggingConfiguration throws exception instead of just logging failure.

### ISSUE THREAD

**Issue by dalmuti509** [NLog/NLog#2878]
**NLog version**: 4.5.9

**Platform**: .Net 4.6.1

**Current NLog config** (xml or C#, if relevant)
None

- What is the current result?
 NLog.LogFactory.GetLogger throws an exception if it can't find an nlog.config file.

- What is the expected result?
No exception, just no configuration loaded.  "Failed loading config from ..." logged to InternalLogger.

- Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)?
Yes

- Please post full exception details (message, stacktrace, inner exceptions)
Note: Private details changed to ... .
``` C#
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. 
---> System.TypeInitializationException: The type initializer for '...' threw an exception. 
---> System.TypeInitializationException: The type initializer for '...' threw an exception. 
---> System.TypeInitializationException: The type initializer for '...' threw an exception. 
---> System.IO.FileNotFoundException: Could not find file '...NLog.config'.
    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   at System.Xml.XmlDownloadManager.GetStream(Uri uri, ICredentials credentials, IWebProxy proxy, RequestCachePolicy cachePolicy)
   at System.Xml.XmlUrlResolver.GetEntity(Uri absoluteUri, String role, Type ofObjectToReturn)
   at System.Xml.XmlTextReaderImpl.FinishInitUriString()
   at System.Xml.XmlTextReaderImpl..ctor(String uriStr, XmlReaderSettings settings, XmlParserContext context, XmlResolver uriResolver)
   at System.Xml.XmlReaderSettings.CreateReader(String inputUri, XmlParserContext inputContext)
   at System.Xml.XmlReader.Create(String inputUri, XmlReaderSettings settings, XmlParserContext inputContext)
   at System.Xml.XmlReader.Create(String inputUri)
   at NLog.Config.XmlLoggingConfiguration.CreateFileReader(String fileName)
   at NLog.Config.XmlLoggingConfiguration..ctor(String fileName, Boolean ignoreErrors, LogFactory logFactory)
   at NLog.Config.XmlLoggingConfiguration..ctor(String fileName, LogFactory logFactory)
   at NLog.LogFactory.TryLoadLoggingConfiguration(String configFile)
   at NLog.LogFactory.TryLoadFromFilePaths()
   at NLog.LogFactory.get_Configuration()
   at NLog.LogFactory.GetLogger(LoggerCacheKey cacheKey)
   at NLog.LogFactory.GetLogger(String name)
   at NLog.LogManager.GetLogger(String name)
   at ...  

```

- Are there any workarounds? yes
Add nlog.config that is empty.

- Is there a version in which it did work?
Maybe 4.5.7... removed a bad nlog.config file from a dependency that was being copied into the folder.

- Can you help us by writing an unit test?
``` C#
[Test]
        public void LogManager_TryLoadLoggingConfiguration_WhenFileDoesNotExist_ThenDoesNotThrowException()
        {
            //Assign
            string filename = $"{Guid.NewGuid()}.config";
            File.Delete(filename);

            //Act
            NLog.LogFactory.TryLoadLoggingConfiguration(filename);            
        }
```

**Comment by snakefoot** [comment]
Strange as there is File.Exists-check:

https://github.com/NLog/NLog/blob/9559441b3fa7df0fca3bfcaec8e9087569a8bf86/src/NLog/LogFactory.cs#L358-L360

But yes there should be a try-catch. It should not fail when scanning random location, where is access might be denied.

**Comment by 304NotModified** [comment]
fixed by #2883 :) (thanks @snakefoot)
